### ISSUE TITLE
False Negative: RV_RETURN_VALUE_IGNORED_NO_SIDE_EFFECT missed for java.util.Arrays.copyOf

### ISSUE THREAD

**Issue by leemeii** [spotbugs/spotbugs#3900]
**Describe the bug**
The rule RV_RETURN_VALUE_IGNORED_NO_SIDE_EFFECT fails to flag cases where the return value of java.util.Arrays.copyOf() is ignored.

**To Reproduce**

```java
package test;

import java.util.Arrays;

public class Bug {
    public void test() {
        int[] arr = {1, 2, 3};
        Arrays.copyOf(arr, 10); <-  should report (FN)
    }
}
```

**Expected behavior**
SpotBugs should report RV_RETURN_VALUE_IGNORED_NO_SIDE_EFFECT on Arrays.copyOf, which has no side effect and returns a newly allocated array, wasting both CPU cycles and memory.

**Actual behavior**
False Negative 

**Environment**
- Java version: JDK 17
- SpotBugs Version: 4.9.8
- How do you use SpotBugs: CLI

**Comment by leemeii** [comment]
The issue stems from an incomplete "Side-Effect-Free" method whitelist in the source code.

In edu.umd.cs.findbugs.detect.FindNoSideEffectMethods.java, the detection logic for java/util/Arrays is hardcoded but only includes a limited set of methods:
```
Java

// Snippet from FindNoSideEffectMethods.java
if ((methodName.equals("binarySearch") && (className.equals("java/util/Arrays") || className.equals("java/util/Collections")))
        || methodName.startsWith("$SWITCH_TABLE$")
        || ... ) {
    return true;
}
```
While binarySearch is correctly identified, copyOf is missing from this list.

**Comment by JuditKnoll** [comment]
Thank you for the report and even finding the location of the issue @leemeii 
Are you interested in opening a PR fixing this issue?

**Comment by leemeii** [comment]
> Thank you for the report and even finding the location of the issue [@leemeii](https://github.com/leemeii) Are you interested in opening a PR fixing this issue?

Hi @JuditKnoll, thank you for the feedback. I will prepare a PR to update the whitelists and ensure these java.util.Arrays methods are correctly recognized as pure functions.
